src/main/java/com/google/api/codegen/config/DiscoveryField.java

```diff
@@ -123,6 +135,11 @@ public boolean isRepeated() {
   }
 
   @Override
+  public boolean mayBeInResourceName() {
```

Comment:
Could you explain what this means?

Reply:
If the result is true, then it is possible that the Field is in a generated ResourceName. If the result is false, then that Field will never be in a generated ResourceName.

This is necessary because a ResourceNameTreatment in the gapic config applies to a whole method, but some Discovery methods have path parameter Fields that can go in the ResourceName in addition to a request body object that doesn't go in the ResourceName. Then the request body Field should return false in mayBeInResourceName() so that it doesn't get thrown in the ResourceName.

Reply:
Also added a comment in the implementation to help explain.